### PR TITLE
v23 compatibility

### PR DESCRIPTION
**Pull request by benallfree**
This is a PR started for v23 compatibility.

## Changes
* Schema now just installs the custom tables needed for testing, not the stock tables.
* Updated tests with new v23 error message text
* Always start test server in `--dev` mode
* Update REST API paths where necessary
* Reference PocketBase v23
* Removed deprecated "external auth" APIs
* Creating a post with no body now works, because there are no required fields

## Enhancements

* Introduced a `Get` method to fetch custom endpoints. If this is good, I'll add a `Post`, `Patch`, `Put`, and `Delete` as well.
* Renamed `WithDebug()` to `WithRestDebug()`
* Added a `WithSseDebug()` that logs raw SSE events 
* Added a `REST_DEBUG` env var to enable `WithRestDebug()` without recompiling
* Added an `SSE_DEBUG` env var to enable `WithSseDebug()`  without recompiling

## v22 backward compat

I'm not sure exactly how to handle all of this, but I'm trying to make it so the library can talk to both v22 and v23 PocketBase APIs. To do that, I added a `22` suffix to methods that are incompatible with v23:

* `WithAdminEmailPassword22`
* `ListAuthMethods22`
* `ListExternalAuths22`
* `UnlinkExternalAuth22`

## v22 test suite

The test suite is currently v23-only, but it could be possible to refactor so it passes against both v22 and v23. But that means maintaining v22 migrations as well, so I think it's better to just drop v22 testing from this version and maintain the methods for backward compat without tests. However, the test migrations seem better suited as test setup/teardown. This could help us keep the test suite so it passes against both v22 and v23.

### CODE REVIEW

File: client_test.go [hunk 1]
```diff
@@ -349,7 +349,8 @@ func TestClient_Create(t *testing.T) {
 			name:       "Create with no body",
 			client:     defaultClient,
 			collection: migrations.PostsPublic,
-			wantErr:    true,
+			wantErr:    false,
+			wantID:     true,
```
Comment (benallfree) [inline client_test.go:353]: This works because there are no required fields. I'm not sure if it worked before or not.

File: Makefile [hunk 2]
```diff
@@ -17,7 +17,7 @@ build: ## Build all executables, located under ./bin/
 
 serve: build ## Run the pocketbase server
 	@echo "Running server..."
-	@./bin/pocketbase serve
+	@./bin/pocketbase serve --dev
```
Comment (benallfree) [inline Makefile:20]: Seems helpful to start the server in dev mode

File: client.go [hunk 3]
```diff
@@ -41,6 +43,15 @@ func NewClient(url string, opts ...ClientOption) *Client {
 	return c
 }
 
+func NewTestClient(url string, opts ...ClientOption) *Client {
```
Comment (benallfree) [inline client.go:46]: This will show REST traffic if `DEBUG=1 make test`

File: client.go [hunk 4]
```diff
@@ -49,7 +60,7 @@ func WithDebug() ClientOption {
 
 func WithAdminEmailPassword(email, password string) ClientOption {
 	return func(c *Client) {
-		c.authorizer = newAuthorizeEmailPassword(c.client, c.url+"/api/admins/auth-with-password", email, password)
+		c.authorizer = newAuthorizeEmailPassword(c.client, c.url+fmt.Sprintf("/api/collections/%s/auth-with-password", core.CollectionNameSuperusers), email, password)
```
Comment (benallfree) [inline client.go:63]: Now using the core constant instead of hard-coding

File: client_test.go [hunk 5]
```diff
@@ -377,14 +378,22 @@ func TestClient_Create(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			r, err := tt.client.Create(tt.collection, tt.body)
-			assert.Equal(t, tt.wantErr, err != nil, err)
-			assert.Equal(t, tt.wantID, r.ID != "")
+			if tt.wantErr {
```
Comment (benallfree) [inline client_test.go:381]: This produces much more helpful error messages

File: migrations/1668292995_collections_snapshot.go [hunk 6]
```diff
@@ -3,162 +3,149 @@ package migrations
...
+	"github.com/pocketbase/pocketbase/core"
 	m "github.com/pocketbase/pocketbase/migrations"
-	"github.com/pocketbase/pocketbase/models"
 )
 
 // Auto generated migration with the most recent collections configuration.
 func init() {
-	m.Register(func(db dbx.Builder) error {
+	m.Register(func(app core.App) error {
 		jsonData := `[
```
Comment (benallfree) [inline migrations/1668292995_collections_snapshot.go:13]: This now includes just the custom tables, not the system tables (no need).

File: record.go [hunk 7]
```diff
@@ -3,31 +3,63 @@ package pocketbase
...
+
+	// technically could be omitted if the provider doesn't support PKCE,
+	// but to avoid breaking existing typed clients we'll return them as empty string
+	CodeVerifier        string `json:"codeVerifier"`
+	CodeChallenge       string `json:"codeChallenge"`
+	CodeChallengeMethod string `json:"codeChallengeMethod"`
+}
+
+// Borrowed from https://github.com/pocketbase/pocketbase/blob/844f18cac379fc749493dc4dd73638caa89167a1/apis/record_auth_methods.go#L52
+type AuthMethodsResponse struct {
```
Comment (benallfree) [inline record.go:47]: See https://github.com/pocketbase/pocketbase/issues/6050

File: record.go [hunk 8]
```diff
@@ -381,69 +413,6 @@ func (c *Collection[T]) ConfirmEmailChange(emailChangeToken string, password str
...
-	Created      string `json:"created"`
-	Updated      string `json:"updated"`
-	RecordID     string `json:"recordId"`
-	CollectionID string `json:"collectionId"`
-	Provider     string `json:"provider"`
-	ProviderID   string `json:"providerId"`
-}
-
-// ListExternalAuths lists all linked external auth providers for the specified auth record.
-func (c *Collection[T]) ListExternalAuths(recordID string) ([]ExternalAuthRequest, error) {
```
Comment (benallfree) [inline record.go:395]: Deprecated

File: record_test.go [hunk 9]
```diff
@@ -175,61 +178,29 @@ func TestCollection_ConfirmPassworReset(t *testing.T) {
...
+		defaultClient := NewTestClient(defaultURL)
 
 		err := CollectionSet[User](defaultClient, "users").ConfirmEmailChange("eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJjb2xsZWN0aW9uSWQiOiJfcGJfdXNlcnNfYXV0aCIsImVtYWlsIjoidXNlckB1c2VyLmNvbSIsImV4cCI6MTcxMzQ3MTc5NSwiaWQiOiI4dng5aHVkNmRlcDIydXYiLCJ0eXBlIjoiYXV0aFJlY29yZCJ9.u_7_u1t0MueFfKAMmXPqe4o1mNBn_-oFEpdSSeGqlUs",
 			"user@user.com")
 		assert.Error(t, err)
 		assert.Contains(t, err.Error(), "validation_invalid_token")
 	})
 }
-
-func TestCollection_ListExternalAuthMethods(t *testing.T) {
```
Comment (benallfree) [inline record_test.go:205]: Deprecated

**Comment by pluja** [comment]
Awesome

**Comment by benallfree** [comment]
Great, I have a stack of subsequent PRs, would you mind if I opened one at a time so we can discuss and merge?

**Comment by pluja** [comment]
no problem, go ahead!

**Comment by benallfree** [comment]
@pluja By the way, it looks like this PR was merged instead of squash-merged and it made a bunch of `wip` commits in the main branch. I recommend undoing that last marge commit and force-pushing to clean up the commit history before proceeding :) 

Also, if you want to make me a maintainer I have some spare time to work on this.

**Comment by pluja** [comment]
@benallfree I sent you an invitation, thanks!!

**Pull request merged**